### ISSUE TITLE
Browser validator should show & explain the 'upload' warning message that users see

### ISSUE THREAD

**Issue by mekline** [psych-ds/validator-prototype#7]
Even though the validator is actually fully client side, your browser asks you if you really want to upload files (see screenshot). We should show people what that message looks like, and why it's still safe to click 'okay' and will not be sending your files onto a server somewhere.

(The current message is intelligible to people who already understand the client side concept, probably not otherwise.

![Screen Shot 2019-09-19 at 1 23 45 PM](https://user-images.githubusercontent.com/6819829/65266405-fe91ac80-dae0-11e9-9602-18f9f59fde87.png)

**Comment by FelixHenninger** [comment]
Thanks Melissa, that's a fantastic catch! I hadn't seen this warning before, so I played around a while, and I couldn't reproduce it unless I uploaded my documents folder by clicking on the validator button and selecting the documents folder (which I assume was what you did?). Even if I drag-and-drop my documents folder, I don't get the same message (though the browser slows to a crawl, which is another issue probably). Even other large folders, e.g. my downloads, work without a warning.

So this left me wondering: Is this something that regular users will see? I can't figure out from the [chromium source code](https://chromium.googlesource.com/chromium/src.git/+/997810155defb24013257791f46dd1c50310cfc2/chrome/browser/ui/views/folder_upload_confirmation_view.cc) what triggers the warning, though in their bug tracker they say that [they're trying to prevent malicious sites from tricking the user into uploading their documents folder](https://bugs.chromium.org/p/chromium/issues/detail?id=637098), so I'm guessing that they're somehow testing which folder is uploaded. That being said, I've 'uploaded' a lot to the validator prototype by now, so maybe the browser has picked up on that and won't show me the warning any more?

Finally, this also reminds me that I wanted to remove the 'click to select' text, because the input via the file selector is prone to buginess compared to drag-and-drop. I'll do that over the coming days!

What do you think, and have you seen this warning in your regular use of the validator? I'm traveling and don't have access to other computers, but if you do and can spare a moment, could you test what happens when users use the validator for the very first time?

Best, and many thanks again,


-Felix

**Comment by mekline** [comment]
Huh - I tried a few things, and it seems to matter whether I drag-and-drop
or click to select (the latter triggers the warning, even for a folder that
isn't my 'Documents' folder.)

We'll need a warning about the browser/explanation what client side means
in any case, since people are likely to be nervous about doing stuff
in-browser. But if killing click-to-select eliminates the confusing warning
in many cases, that's good!

**Comment by FelixHenninger** [comment]
Hej Melissa, thanks for the more thorough testing! It's weird that the input modality affects the warning 🤔. Anyhow, one more reason to deprecate it (the other is that the results are inconsistent, as reported by Peder in #2).

My suggestion would be to leave the click-to-select in place, and fix the edge cases in the mid to long term, but simply to remove the label that invites folks to click-to-select, and ask only for drag-and-drop input.

**Comment by FelixHenninger** [comment]
(ok, just as a quick update, I finally managed to push my commit removing the click-to-upload instruction at 2eca8d46e73749b834e28b7eb4306803ca3f0dc7)

**Comment by mekline** [comment]
For the warning message display, I think this is just text/photos we can put (or link to) in the sidebar. Should we merge this with #6?

**Comment by FelixHenninger** [comment]
Hej @mekline, thanks! Yes, I think this is closely related with the sidebar issue. That being said, I would love to see independent confirmation of the error message popping up in regular use, just so that we can judge whether this is a frequent occurance / which weight to give this in the sidebar.

@pederisager, @lindemann09, @axrhart, if you have a moment, could you please check whether you can re-create the warning in the screenshot above in chrome by (a) checking an example dataset or (b) 'uploading' your home directory. I think it would be super-useful to know whether this is something users will run into.

Thanks y'all! -F

**Comment by pederisager** [comment]
I don't currently get this warning but I think that's because I permanently changed some browser setting. I definitely used to get this same message for uploaded directories.

**Comment by lindemann09** [comment]
@FelixHenninger 
Yes, I get this warning using Chromium and Google's Chrom
![image](https://user-images.githubusercontent.com/2202854/67197610-76633780-f3fd-11e9-94c4-9688a75956a0.png)

**Comment by axrhart** [comment]
@FelixHenninger I can't reproduce this warning with an untouched Chromium (71.0.3578.98; no add-ons; default user profile) uploading the local validator-web folder with about 2k files (my home folder is too large to get any result). The folder and test cases returns a validation.

Felix, if you can't find anything in the source code, could this be a Google module that installs independently or function when it is connected to your user account?

**Comment by lindemann09** [comment]
@FelixHenninger @axrhart  Note, I get the warning only, if I click and select a folder. Drag and drop does not cause a warning.

**Comment by axrhart** [comment]
Good point @lindemann09 . Confirmed in my Chromium for both, the 3 files folder and the 2088 files folder as long as I use the standard file upload route via click and select.

So detaching the click event handler might solve this?
